commit a18190903665d59bdf3289094017d984159907ac
Author: Tim Foster <tim.foster@joyent.com>
Date:   2019-08-14T22:25:52+01:00 (8 weeks ago)
    
    OS-7921 SmartOS bash_completions should complete alias names (fix script)
    Reviewed by: Mike Gerdts <mike.gerdts@joyent.com>

diff --git a/usr/src/cmd/nsadmin/bash/bash_completion.d/zone_alias b/usr/src/cmd/nsadmin/bash/bash_completion.d/zone_alias
index 75de543b08..b0a465060f 100644
--- a/usr/src/cmd/nsadmin/bash/bash_completion.d/zone_alias
+++ b/usr/src/cmd/nsadmin/bash/bash_completion.d/zone_alias
@@ -4,7 +4,7 @@ _zone_alias()
     alias="${COMP_WORDS[COMP_CWORD]}"
     RES=$(vmadm list -H -o alias,uuid | awk -v alias="$alias" '
         BEGIN { count = 0; uuid = ""; }
-        $1 ~ "^"alias" " { count = 1; aliases[1] = $1; uuid = $2; exit;}
+        $1 == alias { count = 1; aliases[1] = $1; uuid = $2; exit;}
         $1 ~ "^"alias { count += 1; aliases[count] = $1; uuid = $2; }
         END { if (count == 1) {
                   print "smartos_alias_completed " aliases[1] " " uuid;
